pyth-rng/src/command/run.rs

```diff
@@ -42,34 +44,52 @@ pub async fn run(opts: &RunOptions) -> Result<(), Box<dyn Error>> {
     )]
     struct ApiDoc;
 
-    let contract = Arc::new(PythContract::from_opts(&opts.ethereum).await?);
-    let provider_info = contract.get_provider_info(opts.provider).call().await?;
+    let config = opts.config.load()?;
 
-    // Reconstruct the hash chain based on the metadata and check that it matches the on-chain commitment.
-    // TODO: we should instantiate the state here with multiple hash chains.
-    // This approach works fine as long as we haven't rotated the commitment (i.e., all user requests
-    // are for the most recent chain).
-    let random: [u8; 32] = provider_info.commitment_metadata;
-    let chain = PebbleHashChain::from_config(&opts.randomness, random)?;
-    let chain_state = HashChainState {
-        offsets:     vec![provider_info
-            .original_commitment_sequence_number
-            .try_into()?],
-        hash_chains: vec![chain],
-    };
+    let mut chains = HashMap::new();
+    for chain_config in &config.chains {
+        let contract = Arc::new(PythContract::from_config(&chain_config)?);
+        let provider_info = contract.get_provider_info(opts.provider).call().await?;
+
+        // Reconstruct the hash chain based on the metadata and check that it matches the on-chain commitment.
+        // TODO: we should instantiate the state here with multiple hash chains.
+        // This approach works fine as long as we haven't rotated the commitment (i.e., all user requests
+        // are for the most recent chain).
+        // TODO: we may want to load the hash chain in a lazy/fault-tolerant way. If there are many blockchains,
+        // then it's more likely that some RPC fails. We should tolerate these faults and generate the hash chain
+        // later when a user request comes in for that chain.
+        let random: [u8; 32] = provider_info.commitment_metadata;
+        let hash_chain =
+            PebbleHashChain::from_config(&opts.randomness, &chain_config.chain_id, random)?;
+        let chain_state = HashChainState {
+            offsets:     vec![provider_info
+                .original_commitment_sequence_number
+                .try_into()?],
+            hash_chains: vec![hash_chain],
+        };
+
+        if chain_state.reveal(provider_info.original_commitment_sequence_number)?
+            != provider_info.original_commitment
+        {
+            return Err(anyhow!(format!("The root of the generated hash chain for chain id {} does not match the commitment. Are the secret and chain length configured correctly?", &chain_config.chain_id)).into());
```

Comment:
`anyhow!` already formats, so you can do `anyhow!("Woah {}", 1);` over `anyhow!(format!("Woah {}", 1));`